test/spawn.jl

```diff
@@ -428,3 +428,11 @@ if is_unix()
         end
     end
 end
+
+# Test for PR 17803
+let p=Pipe()
+    Base.link_pipe(p; julia_only_read=true, julia_only_write=true)
+    ccall(:jl_static_show, Void, (Ptr{Void}, Any), p.in, "world")
+    @async close(p.in)
+    @test readstring(p.out) == "\"world\""
```

Comment:
the buffering is done by the writer, so it's `~2^31` minimum (on the platforms we support) :P